api/src/test/java/org/pmiops/workbench/billing/FreeTierBillingServiceTest.java

```diff
@@ -524,6 +525,28 @@ public void checkFreeTierBillingUsage_dbUpdate() throws MessagingException {
     assertThat(t1.after(t0)).isTrue();
   }
 
+  // Regression test coverage for RW-8328.
+  @Test
```

Comment:
FYI: the existing tests here were great - I didn't have to modify any of them, and they caught a new regression I accidentally introduced while working on the fix.